docs/Targets.md

```diff
@@ -0,0 +1,123 @@
+# Targets
+
+## SiFive HiFive1 RISC-V
+
+### Features
+* E31 RISC-V 320MHz 32-bit processor
+* Onboard 16KB scratchpad RAM
+* External 4MB QSPI Flash 
+
+### Default Linker Settings
+* FLASH: Address 0x20000000, Len 0x6a120 (424 KB)
+* RAM:   Address 0x80000000, Len 0x4000  (16 KB)
+
+### Stock bootloader
+Start Address: 0x20000000 is 64KB. Provides a "double tap" reset feature to halt boot and allow debugger to attach for reprogramming. Press reset button, when green light comes on press reset button again, then board will flash red.
+
+### Application Code
+Start Address: 0x20010000
+
+### wolfBoot configuration
+
+The default wolfBoot configuration will add a second stage bootloader, leaving the stock "double tap" bootloader for safety.
+
+For testing wolfBoot here are the changes required:
+
+1. Makefile arguments:
+    * ARCH=RISCV
+    * TARGET=hifive1
+ 
+    ```
+    make ARCH=RISCV TARGET=hifive1 clean
+    make ARCH=RISCV TARGET=hifive1
+    ```
+
+    If using the `riscv64-unknown-elf-` cross compiler you can add `CROSS_COMPILE=riscv64-unknown-elf-` to your `make` or modify `arch.mk` as follows:
+    
+    ```
+     ifeq ($(ARCH),RISCV)
+    -  CROSS_COMPILE:=riscv32-unknown-elf-
+    +  CROSS_COMPILE:=riscv64-unknown-elf-
+    ```
+
+
+2. `include/target.h`
+
+Bootloader Size: 0x10000 (64KB)
+
+```c
+#define WOLFBOOT_SECTOR_SIZE                 0x10000
+#define WOLFBOOT_PARTITION_BOOT_ADDRESS      0x20010000
```

Comment:
This is the partition address for the main BOOT image, 0x100 before the application entry point